witchcraft/wdebug/api.go

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) 2025 Palantir Technologies. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package wdebug
+
+import (
+	"context"
+	"io"
+)
+
+type DiagnosticType string
+
+type DiagnosticHandler interface {
+	Type() DiagnosticType
```

Comment:
I know this content is copy-pasted/moved from what already existed, but would be nice to have at least some documentation about the interface and behavior for functions where functionality is non-obvious

Reply:
added documentation